anchore_engine/services/policy_engine/engine/feeds/grypedb_sync.py

```diff
@@ -122,15 +118,31 @@ def _update_grypedb(
                 with GrypeDBStorage() as grypedb_file:
                     with grypedb_file.create_file(active_grypedb.archive_checksum) as f:
                         f.write(grypedb_document)
-                    GrypeWrapperSingleton.get_instance().init_grype_db_engine(
-                        grypedb_file.path,
-                        active_grypedb.archive_checksum,
-                        active_grypedb.schema_version,
-                    )
+                    cls._stage_and_update_grypedb(active_grypedb, grypedb_file.path)
         except Exception as e:
             logger.exception("GrypeDBSyncTask failed to sync")
             raise GrypeDBSyncError(str(e)) from e
 
+    @staticmethod
+    def _stage_and_update_grypedb(
+        active_grypedb: GrypeDBFeedMetadata, grypedb_file_path: Optional[str] = None
+    ):
+        # Stage the new grype_db
+        engine_metadata = GrypeWrapperSingleton.get_instance().stage_grype_db_update(
+            grypedb_file_path,
+            active_grypedb.archive_checksum,
+            active_grypedb.schema_version,
+        )
+        archive_checksum = engine_metadata.archive_checksum
+
+        # Verify that the checksum matches what was expected
+        if archive_checksum == active_grypedb.checksum:
+            # If so, Promote
+            GrypeWrapperSingleton.get_instance().update_grype_db(archive_checksum)
+        else:
+            # Otherwise unstage
+            GrypeWrapperSingleton.get_instance().unstage_grype_db()
+
```

Comment:
@dspalmer99 I think that this needs to raise a GrypeDBSyncError exception. This seems like a failure path that will cause the local instance of policy engine to not actually be running the grypedb that is specified as active within the db. It would happen silently and images could be analyzed with incorrect vuln data